core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -104,11 +106,12 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
    * Get a ShuffleWriterGroup for the given map task, which will register it as complete
    * when the writers are closed successfully
    */
-  def forMapTask(shuffleId: Int, mapId: Int, numBuckets: Int, serializer: Serializer,
-      writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
+  def forMapTask(shuffleId: Int, mapId: Int, stageAttemptId: Int, numBuckets: Int,
+      serializer: Serializer, writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
```

Comment:
Yeah, I don't have a solid sense of when it makes sense to use ShuffleIdAndAttempt and when it is better to use two separate Ints.  Part of me wants to say that since ShuffleIdAndAttempt exists, we should generally use it to convey the nature of the grouping of the two values and steer people away from mistakenly treating them as independent of each other.  On the other hand, their use often seems to contribute nothing but clutter.